central/pruning/pruning.go

```diff
@@ -485,6 +494,140 @@ func (g *garbageCollectorImpl) collectImages(config *storage.PrivateConfig) {
 	}
 }
 
+func (g *garbageCollectorImpl) collectClusters(config *storage.PrivateConfig) {
+	if !features.DecommissionedClusterRetention.Enabled() {
+		return
+	}
+
+	// Check to see if pruning is enabled
+	clusterRetention := config.GetDecommissionedClusterRetention()
+	retentionDays := int64(clusterRetention.GetRetentionDurationDays())
+	if retentionDays == 0 {
+		log.Info("[Cluster Pruning] pruning is disabled.")
+		return
+	}
+
+	// Check to see if enough time has elapsed to run again
+	if lastClusterPruneTime.Add(clusterGCFreq).After(time.Now()) {
+		// Only cluster pruning if it's been at least clusterGCFreq since last run
+		return
+	}
+	defer func() {
+		lastClusterPruneTime = time.Now()
+	}()
+
+	// Allow 24hrs grace period after the config changes
+	lastUpdateTime, err := types.TimestampFromProto(clusterRetention.GetLastUpdated())
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	if timeutil.TimeDiffDays(time.Now(), lastUpdateTime) < 1 {
+		// Allow 24 hr grace period after a change in config
+		log.Info("[Cluster Pruning] skipping pruning to allow 24 hr grace period after a recent change in cluster retention config.")
+		return
+	}
+
+	// Retention should start counting _after_ the config is created (which is basically when upgraded to 71)
+	configCreationTime, err := types.TimestampFromProto(clusterRetention.GetCreatedAt())
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	if timeutil.TimeDiffDays(time.Now(), configCreationTime) < int(retentionDays) {
+		// In this case, the deployments that became unhealthy after config creation would also be unhealthy for fewer than retention days
+		// and pruning can be skipped
+		log.Info("[Cluster Pruning] skipping pruning as retention period only starts after upgrade to version 3.71.0.")
+		return
+	}
+
+	allClusterCount, err := g.clusters.CountClusters(pruningCtx)
+	if err != nil {
+		log.Errorf("[Cluster Pruning] error counting clusters: %v", err)
+	}
+
+	if allClusterCount == 0 {
+		log.Info("[Cluster Pruning] skipping pruning because no clusters were found.")
+		return
+	}
+
+	query := search.NewQueryBuilder().
+		AddDays(search.LastContactTime, retentionDays).
+		AddExactMatches(search.SensorStatus, storage.ClusterHealthStatus_UNHEALTHY.String()).ProtoQuery()
+	clusters, err := g.clusters.SearchRawClusters(pruningCtx, query)
```

Comment:
You do not need to pull the full cluster object for ID. You can use `Search` which returns `search.Result` with ID field.

Reply:
Need the cluster object here as I need to go though the cluster labels to find an intersection with ignoreLables.